src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -125,13 +125,17 @@ export function createRecentNavLink(
   recentLink: ChromeRecentlyAccessedHistoryItem,
   navLinks: ChromeNavLink[],
   basePath: HttpStart['basePath'],
-  navigateToUrl: InternalApplicationStart['navigateToUrl']
+  navigateToUrl: InternalApplicationStart['navigateToUrl'],
+  workspaceEnabled: boolean = true
 ): RecentNavLink {
   const { link, label, workspaceId } = recentLink;
   const href = relativeToAbsolute(
-    basePath.prepend(formatUrlWithWorkspaceId(link, workspaceId || '', basePath), {
-      withoutClientBasePath: true,
-    })
+    basePath.prepend(
+      formatUrlWithWorkspaceId(link, workspaceEnabled ? workspaceId || '' : '', basePath),
```

Comment:
This logic looks a little bit strange to me, I think we don't need to call `formatUrlWithWorkspaceId` if workspaceEnabled equal false. How about refactor with below code?
```suggestion
      workspaceEnabled ? formatUrlWithWorkspaceId(link, workspaceId || '', basePath) : link,
```
By the way, can we add some unit tests about it?

Reply:
sure